flang/include/flang/Optimizer/Dialect/FIROps.td

```diff
@@ -81,9 +81,13 @@ def AnyBoxLike : TypeConstraint<Or<[fir_BoxType.predicate,
 def fir_ShapeType : Type<CPred<"$_self.isa<fir::ShapeType>()">, "shape type">;
 def fir_ShapeShiftType : Type<CPred<"$_self.isa<fir::ShapeShiftType>()">,
     "shape shift type">;
+def fir_ShiftType : Type<CPred<"$_self.isa<fir::ShiftType>()">, "shift type">;
```

Comment:
We should coordinate this with Valentin's work on shuffling the types around.

Reply:
Yes, I'll wait to know if we have to drop the `fir_` and update the patch to the new way of defining types.

Reply:
I kept the `fir_`, but moved the type definition to tablegen now that Valentine's work is in.